tests/org.palladiosimulator.dataflow.confidentiality.analysis.tests/src/org/palladiosimulator/dataflow/confidentiality/analysis/DataFlowConfidentialityAnalysisTest.java

```diff
@@ -2,58 +2,99 @@
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.palladiosimulator.dataflow.confidentiality.analysis.AnalysisUtils.TEST_MODEL_PROJECT_NAME;
 import static org.palladiosimulator.dataflow.confidentiality.analysis.AnalysisUtils.assertSequenceElement;
 import static org.palladiosimulator.dataflow.confidentiality.analysis.AnalysisUtils.assertSequenceElements;
+import static org.palladiosimulator.dataflow.confidentiality.analysis.AnalysisUtils.assertUserSequenceElementContent;
+import static org.palladiosimulator.dataflow.confidentiality.analysis.AnalysisUtils.assertSEFFSequenceElementContent;
 
 import java.nio.file.Paths;
 
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 import org.palladiosimulator.dataflow.confidentiality.analysis.sequence.entity.pcm.CallingSEFFActionSequenceElement;
 import org.palladiosimulator.dataflow.confidentiality.analysis.sequence.entity.pcm.CallingUserActionSequenceElement;
 import org.palladiosimulator.dataflow.confidentiality.analysis.sequence.entity.pcm.SEFFActionSequenceElement;
 import org.palladiosimulator.dataflow.confidentiality.analysis.testmodels.Activator;
 
 public class DataFlowConfidentialityAnalysisTest {
-
-    @Test
-    public void testStandaloneAnalysis() {
+	
+	private static StandalonePCMDataFlowConfidentialtyAnalysis analysis;
+	
+	/**
+	 * Initializes the analysis with the BranchingOnlineShop test model and initializes the Analysis
+	 */
+	@BeforeAll
+	public static void initializeAnalysis() {
         final var usageModelPath = Paths.get("models", "BranchingOnlineShop", "default.usagemodel").toString();
         final var allocationPath = Paths.get("models", "BranchingOnlineShop", "default.allocation").toString();
-
-        final DataFlowConfidentialityAnalysis analysis = new StandalonePCMDataFlowConfidentialtyAnalysis(
+        
+        analysis = new StandalonePCMDataFlowConfidentialtyAnalysis(
                 TEST_MODEL_PROJECT_NAME, Activator.class, usageModelPath, allocationPath);
 
         analysis.initalizeAnalysis();
+	}
+
+	/**
+	 * Tests the standalone analysis by executing the ActionSequenceFinder. Fails if no sequences can be found
+	 */
+    @Test
+    public void testStandaloneAnalysis() {
```

Comment:
I think it would be simpler and better organised to group the test models by functionality, because that would allow better debugging when a test fails, as functionality is grouped together. Some tests for edge cases the Label Propagation encounters would also be better organised in an grouping by functionality as some edge cases only occur in some models.

In return, the setup method for the analysis would not work anymore as it initializes one model, and testing by functionality in each class would require every model to be initialised.

Reply:
I agree, lets's group the tests by functionality then.

Reply:
If you take a look in the recent commits, i have implemented the test for the SequenceFinder in a test class. Currently, i am using a separate method to test each model, but parameterized testing would be possible. 

In implementing that for the `testCount` method however, i have realized that Eclipse does not run parameterized tests and does not show the test results. Should we continue using Parameterized Tests or change them to separate methods?